e2e-tests/nextbutton.test.ts

```diff
@@ -5,87 +5,50 @@ test.describe("Button component is visible on multiple pages", () => {
   let page;
 
   test.beforeAll(async () => {
-    browser = await chromium.launch({ headless: false, slowMo: 3000 });
+    browser = await chromium.launch();
+    page = await browser.newPage();
   });
 
   test.afterAll(async () => {
     await browser.close();
   });
 
   test.beforeEach(async () => {
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/joke-randomiser");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/card-flip-game");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/rate-joke");
-    page = await browser.newPage();
-    await page.goto("http://localhost:5173/joke-fetch");
+    await page.goto("http://localhost:5173");
   });
 
-  test.afterEach(async () => {
-    await page.close();
-  });
+  // Define a reusable function to navigate to pages and check buttons
+  async function navigateToPageAndCheckButton(pageUrl) {
```

Comment:
This is great so does this function allow you to test each next button clicks on every page?

Reply:
Yes, it removes a lot of lines of repeated code, where we were adding a test for each page separately, where now the `navigateToPageAndCheckButton` function will run asynchronously aka not in order and iterates through each page as a combined test, which prevents it from timing out and failing also 🧪